ui-v2/src/components/artifacts/artifacts-filter.tsx

```diff
@@ -0,0 +1,107 @@
+import { SearchInput } from "@/components/ui/input";
+import {
+	Select,
+	SelectContent,
+	SelectItem,
+	SelectTrigger,
+	SelectValue,
+} from "@/components/ui/select";
+import useDebounce from "@/hooks/use-debounce";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { Typography } from "../ui/typography";
+import { filterType } from "./types";
+
+interface ArtifactsFilterProps {
+	filters: filterType[];
+	onFilterChange: (newFilters: filterType[]) => void;
+	totalCount: number;
+}
+
+const artifactTypeOptions = [
+	{ value: "all", label: "All Types" },
+	{ value: "markdown", label: "Markdown" },
+	{ value: "progress", label: "Progress" },
+	{ value: "image", label: "Image" },
+	{ value: "table", label: "Table" },
+];
+
+export const ArtifactsFilterComponent = ({
+	filters,
+	onFilterChange,
+	totalCount,
+}: ArtifactsFilterProps) => {
+	const [searchInputValue, setSearchInput] = useState(
+		filters.find((val) => val.id == "name")?.value ?? "",
+	);
+
+	const debouncedSearchInputValue = useDebounce(searchInputValue, 500);
+
+	const changeArtifactName = useCallback(
+		(value: string) => {
+			onFilterChange([
+				...filters.filter((filter) => filter.id !== "name"),
+				{ id: "name", label: "Name", value },
+			]);
+		},
+		[filters, onFilterChange],
+	);
+
+	const changeArtifactType = useCallback(
+		(value: string) => {
+			onFilterChange([
+				...filters.filter((filter) => filter.id !== "type"),
+				{ id: "type", label: "Type", value },
+			]);
+		},
+		[filters, onFilterChange],
+	);
+
+	useEffect(() => {
+		changeArtifactName(debouncedSearchInputValue);
+	}, [debouncedSearchInputValue, changeArtifactName]);
+
+	const typeValue = useMemo(
+		() => filters.find((val) => val.id == "type")?.value ?? undefined,
```

Comment:
```suggestion
		() => filters.find((val) => val.id == "type")?.value,
```
This should already return an `undefined` value